examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -3,197 +3,441 @@
+
+# %%
+# We observe that the number of neighbors has a big impact on the performance of
+# the model. If one has access to (at least some) ground truth labels, it is
+# then important to tune `n_neighbors` accordingly. A convenient way to do so is
+# to explore values for `n_neighbors` of the order of magnitud of the expected
+# contamination.
+
+# %%
+from sklearn.preprocessing import MinMaxScaler, SplineTransformer, StandardScaler
+
+preprocessor_list = [
+    None,
+    RobustScaler(),
+    StandardScaler(),
+    MinMaxScaler(),
+    SplineTransformer(),
+]
+expected_anomaly_fraction = 0.02
+lof = LocalOutlierFactor(n_neighbors=int(n_samples * expected_anomaly_fraction))
+
+fig, ax = plt.subplots()
+for model_idx, preprocessor in enumerate(preprocessor_list):
+    model = make_pipeline(preprocessor, lof)
+    model.fit(X)
+    y_pred = model[-1].negative_outlier_factor_
+    display = RocCurveDisplay.from_predictions(
+        y,
+        y_pred,
+        pos_label=pos_label,
+        name=str(preprocessor).split("(")[0],
+        ax=ax,
+        plot_chance_level=(model_idx == len(preprocessor_list) - 1),
+        chance_level_kw={"linestyle": ":"},
+    )
+_ = ax.set_title("Fixed n_neighbors with varying preprocessing\non forestcover dataset")
+
+# %%
+# On the one hand, :class:`~sklearn.preprocessing.RobustScaler` scales each
+# feature independently by using the interquartile range (IQR) by default, which
+# is the range between the 25th and 75th percentiles of the data. It centers the
+# data by subtracting the median and then scale it by dividing by the IQR. The
+# IQR is robust to outliers: the median and interquartile range are less
+# affected by extreme values than the range, the mean and the standard
+# deviation. Furthermore, :class:`~sklearn.preprocessing.RobustScaler` does not
+# squash marginal outlier values, contrary to
+# :class:`~sklearn.preprocessing.StandardScaler`.
+#
+# On the other hand, :class:`~sklearn.preprocessing.MinMaxScaler` scales each
+# feature individually such that its range maps into the range between zero and
+# one. If there are outliers in the data, they can skew it towards either the
+# minimum or maximum values, leading to a completely different distribution of
+# data with large marginal outliers: all non-outlier values can be collapsed
+# almost together as a result.
+#
+# We also evaluated no preprocessing at all (by passing `None` to the pipeline),
+# :class:`~sklearn.preprocessing.StandardScaler` and
+# :class:`~sklearn.preprocessing.SplineTransformer`. Please refer to their
+# respective documentation for more details.
+#
+# Note that the optimal preprocessing depends on the dataset, as shown below:
+
+# %%
+X = X_cardiotocography
+y = y_true["cardiotocography"]
+
+n_samples, expected_anomaly_fraction = X.shape[0], 0.025
+lof = LocalOutlierFactor(n_neighbors=int(n_samples * expected_anomaly_fraction))
+
+fig, ax = plt.subplots()
+for model_idx, preprocessor in enumerate(preprocessor_list):
```

Comment:
I think that the green line will be difficult to distinguish from the red for colorblind. Do you mind to also have a different linestyle for each line to distinguish them this way as well.